### DISCUSSION TITLE
part-ii/2017-24/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#395] [Question comments]
Part II 2017 Automata and Formal Languages: Paper 3, Section I, 4H 

https://questions.tripos.org/part-ii/2017-24/

**Comment by artie2000** [comment]
All bookwork or on example sheets.

<details>
<summary>For the last part, use</summary>

(the complement of) $\{ww\mid w\in\{a,b\}^\star\}$.

</details>
